cpp/src/Slice/Parser.h

```diff
@@ -395,10 +395,14 @@ namespace Slice
         /// Returns the mapped identifier that this element will use in the target language.
         [[nodiscard]] std::string mappedName() const;
         /// Returns the mapped scope that this element will be generated in in the target language.
-        /// (equivalent to `mappedScope(separator) + mappedName()`).
-        [[nodiscard]] std::string mappedScoped(const std::string& separator = "::") const;
+        /// @param separator this string will be used to separate scope segments (defaults to "::").
+        /// @param leading if true, the returned string will have a leading `separator` (defaults to true).
+        /// @remark equivalent to `mappedScope(separator, leading) + mappedName()`.
+        [[nodiscard]] std::string mappedScoped(const std::string& separator = "::", bool leading = true) const;
```

Comment:
I think we should remove the default separator value. It really only applies to C++. Since most (at least half I think) of the class now explicitly need to pass a separator so they can pass false. I think we should remove the default bool for leading as well.

Reply:
Fair enough about the separator. I removed the default, since like you said, it wasn't used very much.

For the bool parameter, how would you feel about switching the default to `false` instead of `true`?
There's 110 places where we pass `false`, and only 61 places where we pass `true`.
And I think "not having extra stuff at the front" is a reasonable and expected behavior to get by default.
Or are you just against the idea of a default bool here in general?